aws/cloudformation/iam.yml.erb

```diff
@@ -835,6 +840,309 @@ roles.each do |name, config| %>
+            Condition:
+              StringEquals:
+                # See: https://docs.github.com/en/actions/security-for-github-actions/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services#configuring-the-role-and-trust-policy
+                token.actions.githubusercontent.com:aud: sts.amazonaws.com
+                token.actions.githubusercontent.com:sub: 'repo:code-dot-org/code-dot-org:environment:marketing-sites-<%=environment_type%>'
+      Policies:
+        - PolicyName: github-actions-cloudformation-policy
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+              - Sid: CloudFormationPermissions
+                Effect: Allow
+                Action:
+                  - cloudformation:CreateStack
+                  - cloudformation:DescribeStacks
+                  - cloudformation:CreateChangeSet
+                  - cloudformation:DescribeChangeSet
+                  - cloudformation:DeleteChangeSet
+                  - cloudformation:ExecuteChangeSet
+                  - cloudformation:ListStackResources
+                Resource: '*'
+              - Sid: PassRoleToCloudFormation
+                Effect: Allow
+                Action:
+                  - iam:PassRole
+                Resource: !GetAtt CloudFormationServiceMarketingSites<%=environment_type.capitalize%>Role.Arn
+
+  # Enforce least privilege on Roles created/passed/attached by the Role that provisions marketing site Stacks.
+  MarketingSitesRoleCreationBoundary<%=environment_type.capitalize%>Policy:
+    Type: AWS::IAM::ManagedPolicy
+    Properties:
+      Description: Permissions boundary for Roles created by CloudFormationMarketingSites<%=environment_type.capitalize%>Role
+      ManagedPolicyName: marketing-sites-role-permissions-boundary-<%=environment_type%>
+      PolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Sid: AllowSecretsManagerForMarketingSites
+            Effect: Allow
+            Action:
+              - secretsmanager:GetSecretValue
+              - secretsmanager:DescribeSecret
+            Resource: 'arn:aws:secretsmanager:*:*:secret:marketing-sites/<%=environment_type%>/*'
+          - Sid: AllowCloudfrontInvalidation
+            Effect: Allow
+            Action:
+              - cloudfront:CreateInvalidation
+            Resource: '*'
+            Condition:
+              StringEquals:
+                aws:ResourceTag/environment-type: <%=environment_type%>
+
+          # Add more statements here as we need the Roles provisioned within a marketing sites Stack to have more permissions.
+
+          # Explicit deny for everything else to enforce least privilege
+          - Sid: DenyAllOtherActions
+            Effect: Deny
+            NotAction:
+              - secretsmanager:GetSecretValue
+              - secretsmanager:DescribeSecret
+              - cloudfront:CreateInvalidation
+            Resource: '*'
```

Comment:
I don't think you need this. I think you only need to put a `Deny` in your permissions boundary policy if you want to remove permissions that this boundary policy would otherwise allow. This policy does not have an `Allow ec2` and therefore any user/role with this boundary applied will not receive ec2 permissions from any policies. Explicitly denying everything not listed in the permissions boundary is redundant.

[This example](https://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies_boundaries.html#access_policies_boundaries-delegate) describes a scenario where the permissions boundary permits `ec2:*`, but denies ec2 actions on a specific instance, but they don't include an explicit `Deny` for RDS, because RDS was never permitted by the boundary in the first place.

Reply:
Good point! Done.